python-package/lightgbm/basic.py

```diff
@@ -41,21 +41,37 @@ def warning(self, msg: str) -> None:
         warnings.warn(msg, stacklevel=3)
 
 
-_LOGGER: Union[_DummyLogger, Logger] = _DummyLogger()
+_LOGGER: Any = _DummyLogger()
+_INFO_METHOD_NAME = "info"
+_WARNING_METHOD_NAME = "warning"
 
 
-def register_logger(logger: Logger) -> None:
+def register_logger(
+    logger: Any, info_method_name: str = "info", warning_method_name: str = "warning"
+) -> None:
     """Register custom logger.
 
     Parameters
     ----------
-    logger : logging.Logger
+    logger : Any
         Custom logger.
+    info_method_name: str, optional (default="info")
+        Method used to log info messages.
+    warning_method_name: str, optional (default="warning")
+        Method used to log warning messages.
```

Comment:
```suggestion
    info_method_name : str, optional (default="info")
        Method used to log info messages.
    warning_method_name : str, optional (default="warning")
        Method used to log warning messages.
```

Reply:
Fixed. Maybe we can add a corresponding linter rule? Or even better, use a unified formatter.

Reply:
Nice idea! Do you know any one?
`pydocstyle` linter we use here doesn't catch such inconsistency.
https://github.com/microsoft/LightGBM/blob/d163c2c16629e3da41dc5d7b8dbb76b8fc6249db/.ci/test.sh#L81

Reply:
Seems this is a [known issue](https://github.com/PyCQA/pydocstyle/issues/519) with pydocstyle. I tried `numpydoc` as mentioned in the comment, however it is not satisfying either. First, `numpydoc` is an extension to `sphinx`, and the validation rules cannot be set in command line options when used as a standalone tool. Second, `numpydoc` is not intelligent enough. We can see this in above issue, where a single missing whitespace will result in multiple false positives.

```
smalltest.no:PR01:Parameters {'a'} not documented
smalltest.no:PR02:Unknown parameters {'a: int'}
```

These two errors are due to parsing error caused by the missing whitespace before colon.

So, I don't have suggested tool suited for this 😞

Reply:
Ah no problem, thanks very much for investigating!